src/main/java/net/minestom/server/ServerFacadeImpl.java

```diff
@@ -0,0 +1,104 @@
+import net.minestom.server.world.biomes.BiomeManager;
+import net.minestom.server.world.biomes.BiomeManagerImpl;
+
+@Getter
+final class ServerFacadeImpl implements ServerFacade {
+
+    private final ExceptionHandler exceptionHandler;
+    private final ConnectionManager connectionManager;
+    private final PacketListenerManager packetListenerManager;
+    private final PacketProcessor packetProcessor;
+    private final InstanceManager instanceManager;
+    private final BlockManager blockManager;
+    private final CommandManager commandManager;
+    private final RecipeManager recipeManager;
+    private final TeamManager teamManager;
+    private final GlobalEventHandler globalEventHandler;
+    private final SchedulerManager schedulerManager;
+    private final BenchmarkManager benchmarkManager;
+    private final DimensionTypeManager dimensionTypeManager;
+    private final BiomeManager biomeManager;
+    private final AdvancementManager advancementManager;
+    private final BossBarManager bossBarManager;
+    private final TagManager tagManager;
+    private final Server server;
+
+    private final ChunkDispatcher chunkDispatcher;
+    private final Ticker ticker;
+
+    private final ServerSettings serverSettings;
+    private final AudienceManager audienceManager;
+    private final MojangAuth mojangAuth;
+    private final ServerStarter serverStarter;
+
+
+    public ServerFacadeImpl(ServerSettings serverSettings) {
+        this.serverSettings = serverSettings;
+        this.exceptionHandler = new ExceptionHandlerImpl();
+        this.globalEventHandler = new GlobalEventHandlerImpl(this);
+        this.packetListenerManager = new PacketListenerManagerImpl(this);
+        this.packetProcessor = new PacketProcessorImpl(this);
+        this.chunkDispatcher = ChunkDispatcher.singleThread(this);
+        this.blockManager = new BlockManagerImpl();
+        this.bossBarManager = new BossBarManagerImpl(this);
+        this.biomeManager = new BiomeManagerImpl();
+        this.instanceManager = new InstanceManagerImpl(this);
+
+        this.commandManager = new CommandManagerImpl(this, this);
+        this.recipeManager = new RecipeManagerImpl();
+
+        this.schedulerManager = new SchedulerManagerImpl();
+        this.benchmarkManager = new BenchmarkManagerImpl(this);
+        this.dimensionTypeManager = new DimensionTypeManagerImpl();
+        this.advancementManager = new AdvancementManagerImpl(serverSettings);
+        this.tagManager = new TagManagerImpl();
+        this.teamManager = new TeamManagerImpl(serverSettings);
+        this.connectionManager = new ConnectionManagerImpl(tagManager, this);
+        this.server = new ServerImpl(serverSettings, this);
+        this.audienceManager = new AudienceManagerImpl(commandManager, this, this);
+        this.ticker = new TickerImpl(this);
+        this.serverStarter = new ServerStarter(this);
+        this.mojangAuth = new MojangAuth(this, this);
```

Comment:
This should not be present unless used, perhaps it would be better to have some generic AuthProvider which this, velocity, bungee implement to prevent the confusion of enabling multiple at once.

Reply:
I agree but I need help with this